controller/AuthoringTool.php

```diff
@@ -87,4 +99,52 @@ public function launch(): void
 
         $this->forward('run', null, null, $_GET);
     }
+
+    /**
+     * @return bool
+     * @throws ContainerExceptionInterface
+     * @throws NotFoundExceptionInterface
+     */
+    private function isFeatureTaoAsToolEnabled(): bool
+    {
+        return $this->getServiceManager()
+            ->getContainer()
+            ->get(FeatureFlagChecker::class)
+            ->isEnabled(FeatureFlagCheckerInterface::FEATURE_FLAG_TAO_AS_A_TOOL);
+    }
+
+    /**
+     * @return LtiMessagePayloadInterface
+     * @throws ContainerExceptionInterface
+     * @throws InterruptedActionException
+     * @throws LtiException
+     * @throws NotFoundExceptionInterface
+     */
+    private function getLtiMessageOrRedirectToLogin(): LtiMessagePayloadInterface
+    {
+        if (!$this->isFeatureTaoAsToolEnabled()) {
+            $this->getLogger()->info(
+                'TAO as tool feature is disabled. The user will be redirected to the login page.'
+            );
+            $this->redirect(_url('login', 'Main', 'tao'));
+        }
+
+        try {
+            $message = $this->getValidatedLtiMessagePayload();
+        } /** @noinspection PhpRedundantCatchClauseInspection */ catch (LtiException $exception) {
+            if ($exception->getMessage() !== self::LTI_NO_MATCHING_REGISTRATION_FOUND_MESSAGE) {
+                throw $exception;
+            }
+
+            $this->getLogger()->info(
```

Comment:
Is this an info or a warning or even error?

Reply:
Now it is an info as this flow is expected by the application

Reply:
Is it expected to throw an error when trying to login into authoring tool?

Reply:
We do not throw an error, but we throw an **exception** (LTIException).

Reply:
yes, in this case, it is expected someone trying to access it without the proper config. in this case what we want to do it to redirect the user to the normal login page.

Reply:
but this is what is being defined by the product and we should reflect what the product says in our code base. So you guys are saying that is not an expected behavior? Can you guys point me out why? Based on what?

Reply:
I would go with a warning on this, it is fine for me, but I do not think it is an error.